vpr/src/draw/draw.cpp

```diff
@@ -4026,6 +4040,219 @@ static void set_force_pause(GtkWidget* /*widget*/, gint /*response_id*/, gpointe
+        invalid_breakpoint_entry_window("x value out of bounds");
+        valid_input = false;
+    }
+    if (to_loc_y < 0 || to_loc_y > int(device_ctx.grid.height())) {
+        invalid_breakpoint_entry_window("y value out of bounds");
+        valid_input = false;
+    }
+    if (to_subtile < 0 || to_subtile > int(device_ctx.grid[to_loc_x][to_loc_y].type->capacity)) {
+        invalid_breakpoint_entry_window("invalid subtile value");
+        valid_input = false;
+    }
+
+    //checks if all entries are full
+    if (std::string(gtk_entry_get_text((GtkEntry*)block_entry)).empty() || std::string(gtk_entry_get_text((GtkEntry*)to_x_entry)).empty() || std::string(gtk_entry_get_text((GtkEntry*)to_y_entry)).empty() || std::string(gtk_entry_get_text((GtkEntry*)subtile_entry)).empty()) {
+        invalid_breakpoint_entry_window("Not all fields are complete");
+        valid_input = false;
+    }
+
+    if (valid_input) {
+        //update draw_manual_move_info
+        manual_move_globals.draw_manual_move_info.valid_input = true;
+        manual_move_globals.draw_manual_move_info.to_x = to_loc_x;
+        manual_move_globals.draw_manual_move_info.to_y = to_loc_y;
+        manual_move_globals.draw_manual_move_info.subtile = to_subtile;
+        manual_move_globals.draw_manual_move_info.block_id = block_id;
+
+        //highlight blocks
+        ClusterBlockId clb_index = ClusterBlockId(block_id);
+        draw_highlight_blocks_color(cluster_ctx.clb_nlist.block_type(clb_index), clb_index);
+        application.refresh_drawing();
+
+        GtkWidget* proceed = find_button("ProceedButton");
+        ezgl::press_proceed(proceed, &application);
+    } else
+        manual_move_globals.draw_manual_move_info.valid_input = false;
+}
+
+//this window displays all move costs and the placer move_outcome and let's the user accept or reject the move
+void cost_summary_window() {
+    //window settings
+    GtkWidget* window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
+    gtk_window_set_position((GtkWindow*)window, GTK_WIN_POS_CENTER);
+    gtk_window_set_title((GtkWindow*)window, "Move Costs");
+
+    //create all widgets
+    GtkWidget* grid = gtk_grid_new();
+    GtkWidget* info = gtk_label_new(NULL);
+    gtk_label_set_markup((GtkLabel*)info, "<b>Move Costs and Outcomes</b>");
+    std::string dc_label = "Delta Cost: " + std::to_string(manual_move_globals.draw_manual_move_info.delta_c);
+    GtkWidget* dc = gtk_label_new(dc_label.c_str());
+    std::string dcn_label = "Delta Cost Norm: " + std::to_string(manual_move_globals.draw_manual_move_info.bb_delta_c);
+    GtkWidget* dcn = gtk_label_new(dcn_label.c_str());
+    std::string dtcn_label = "Delta Timing Cost Norm: " + std::to_string(manual_move_globals.draw_manual_move_info.timing_delta_c);
+    GtkWidget* dtcn = gtk_label_new(dtcn_label.c_str());
+    std::string outcome;
+    manual_move_globals.draw_manual_move_info.placer_move_outcome == MOVE_REJECTED ? outcome = "Rejected" : outcome = "Accepted";
```

Comment:
Better to make a little utility routine in place/move_utils that can convert an e_move_result to a string.  A simple way to do that is to add another enum value NUM_MOVE_RESULTS and make an array of strings 
std::string move_result_strings[NUM_MOVE_RESULTS] = {"rejected", "accepted", "aborted"};
Then make a function that is passed in an enum value and returns a string (say pl_move_type_to_str( )).

Reply:
resolved